app/server/appsmith-plugins/elasticSearchPlugin/src/main/java/com/external/plugins/ElasticSearchPlugin.java

```
@@ -0,0 +1,189 @@
+import org.pf4j.PluginWrapper;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+public class ElasticSearchPlugin extends BasePlugin {
+
+    public ElasticSearchPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Slf4j
+    @Extension
+    public static class ElasticSearchPluginExecutor implements PluginExecutor<RestClient> {
+
+        @Override
+        public Mono<ActionExecutionResult> execute(RestClient client,
+                                                   DatasourceConfiguration datasourceConfiguration,
+                                                   ActionConfiguration actionConfiguration) {
+            final ActionExecutionResult result = new ActionExecutionResult();
+
+            String body = actionConfiguration.getBody();
+
+            final String path = actionConfiguration.getPath();
+            final Request request = new Request(actionConfiguration.getHttpMethod().toString(), path);
+            ContentType contentType = ContentType.APPLICATION_JSON;
+
+            if (isBulkQuery(path)) {
+                contentType = ContentType.create("application/x-ndjson");
+
+                // If body is a JSON Array, instead of ND-JSON, convert it to ND-JSON.
+                if (body.trim().startsWith("[")) {
+                    final StringBuilder ndJsonBuilder = new StringBuilder();
+                    try {
+                        List<Object> commands = objectMapper.readValue(body, ArrayList.class);
+                        for (Object object : commands) {
+                            ndJsonBuilder.append(objectMapper.writeValueAsString(object)).append("\n");
+                        }
+                    } catch (IOException e) {
+                        return Mono.error(e);
```

Could we throw a Plugin Exception here instead of generic Mono.error?

---

Done.